packages/fhir-location-management/src/components/LocationForm/index.tsx

```diff
@@ -111,7 +111,11 @@ const LocationForm = (props: LocationFormProps) => {
   const [form] = Form.useForm();
 
   React.useEffect(() => {
-    form.setFieldsValue({ ...initialValues });
+    // #850 - initial Values would override any values so far fed into  the form,
+    form.setFieldsValue({
```

Comment:
Recommend switching to the Form components initialValues. Like so: https://github.com/opensrp/web/blob/5cf77953cfaa298c10ef2f0290c8e20c9d115ae9/packages/card-support/src/components/DownloadClientData/index.tsx#L156

Reply:
@machariamuguku I see, though initial values need to be a prop, for when editing a location, the the locationForm payload is parsed and supplied to the form as initialValues

Reply:
@p-netm  These still flow down the form as props, could be unified conditionally then provided to the form as initialValues prop

Reply:
Not sure I understand, but we would still have the problem where re-renders in the parent component would erase any form values that had been filled out yet, since it would resupply the initial values.

Reply:
@p-netm I'm suggesting something like this: 

const initialValues = isEdit? { ...initialValues, props.editValues} : initialValues
....
<Form
initialValues={initialFormValues}
----